### PR TITLE
Introducing UE_IPV4_INTERNET/IMS variable

### PR DESCRIPTION
**Pull request by b3c3ll**
{Added} Introducing UE_IPV4_INTERNET/IMS variable and some minor changes to readme and DOCKER_HOST_IP

This one is for master branch which can be applied to the others!

### CODE REVIEW

**Comment by herlesupreeth** [comment]
Thanks a lot for your effort. Will review over the weekend and merge

**Comment by b3c3ll** [comment]
Great

File: .env [hunk 1]
```diff
@@ -119,3 +120,9 @@ METRICS_IP=172.22.0.36
 
 # SRSRAN GNB
 SRS_GNB_IP=172.22.0.37
+
+# UE IPv4 Subnet Range for APN=internet
+UE_IPV4_INTERNET=192.168.100.0/24
+
+# UE IPv4 Subnet Range for APN=ims
+UE_IPV4_IMS=192.168.101.0/24
```
Comment (herlesupreeth) [inline .env:128]: Add a new line here please

File: upf/tun_if.py [hunk 2]
```diff
@@ -27,75 +27,84 @@
...
-			help='TUN interface name e.g. ogstun')
+              required=True,
+              help='TUN interface name e.g. ogstun')
 @click.option('--ipv4_range',
-			required=True,
-			callback=validate_ip_net,
-			help='UE IPv4 Address range in CIDR format e.g. 192.168.100.0/24')
+              required=True,
+              callback=validate_ip_net,
+              help='UE IPv4 Address range in CIDR format e.g. {UE_IPV4_INTERNET}')
```
Comment (herlesupreeth) [inline upf/tun_if.py:56]: Dont change here to {UE_IPV4_INTERNET} as 192.168.100.0/24 is meant to be an example for script users

File: upf/tun_if.py [hunk 3]
```diff
@@ -27,75 +27,84 @@
...
 import subprocess
 import ipaddress
+import os
+UE_IPV4_INTERNET = os.environ['UE_IPV4_INTERNET']
 
 """
 Usage in command line:
 e.g:
-$ python3 tun_if.py --tun_ifname ogstun --ipv4_range 192.168.100.0/24 --ipv6_range 2001:230:cafe::/48
+$ python3 tun_if.py --tun_ifname ogstun --ipv4_range {UE_IPV4_INTERNET} --ipv6_range 2001:230:cafe::/48
```
Comment (herlesupreeth) [inline upf/tun_if.py:36]: Dont change here to {UE_IPV4_INTERNET} as 192.168.100.0/24 is meant to be an example for script users

File: upf/upf.yaml [hunk 4]
```diff
@@ -10,13 +10,13 @@ upf:
       - addr: UPF_IP
         advertise: UPF_ADVERTISE_IP
     subnet:
-      - addr: 192.168.100.1/24
+      - addr: UE_IPV4_INTERNET
```
Comment (herlesupreeth) [inline upf/upf.yaml:13]: Changing here to UE_IPV4_INTERNET is not enough, you would need to substitute this value using `sed` as seen in upf_init.sh

File: upf/upf.yaml [hunk 5]
```diff
@@ -10,13 +10,13 @@ upf:
...
         dev: ogstun
         dnn: internet
       - addr: 2001:230:cafe::1/48
         dev: ogstun
         dnn: internet
-      - addr: 192.168.101.1/24
+      - addr: UE_IPV4_IMS
```
Comment (herlesupreeth) [inline upf/upf.yaml:19]: Changing here to UE_IPV4_IMS is not enough, you would need to substitute this value using `sed` as seen in upf_init.sh

File: smf/smf.yaml [hunk 6]
```diff
@@ -22,13 +22,13 @@ smf:
     pfcp:
       - addr: SMF_IP
     subnet:
-      - addr: 192.168.100.1/24
+      - addr: UE_IPV4_INTERNET
```
Comment (herlesupreeth) [inline smf/smf.yaml:25]: Changing here to UE_IPV4_INTERNET is not enough, you would need to substitute this value using sed as seen in smf_init.sh

File: smf/smf.yaml [hunk 7]
```diff
@@ -22,13 +22,13 @@ smf:
...
         dnn: internet
         dev: ogstun
       - addr: 2001:230:cafe::1/48
         dev: ogstun
         dnn: internet
-      - addr: 192.168.101.1/24
+      - addr: UE_IPV4_IMS
```
Comment (herlesupreeth) [inline smf/smf.yaml:31]: Changing here to UE_IPV4_IMS is not enough, you would need to substitute this value using sed as seen in smf_init.sh

File: .env [hunk 8]
```diff
@@ -5,7 +5,8 @@ MCC=001
 MNC=01
 
 TEST_NETWORK=172.22.0.0/24
-DOCKER_HOST_IP=192.168.1.223
+# Change the DOCKER_HOST_IP to your actual IP Address(Ex. 192.168.1.223), Only if you have Multiple Network Interfaces
+DOCKER_HOST_IP=$(hostname -I)   # 192.168.1.223
```
Comment (herlesupreeth) [inline .env:9]: Remove this change please as its better to leave up to users to change to whatever IP they want

File: .env [hunk 9]
```diff
@@ -19,14 +20,14 @@ PCRF_IP=172.22.0.4
 # SGW
 SGWC_IP=172.22.0.5
 SGWU_IP=172.22.0.6
-SGWU_ADVERTISE_IP=172.22.0.6
+SGWU_ADVERTISE_IP=172.22.0.6    # $DOCKER_HOST_IP
```
Comment (herlesupreeth) [inline .env:23]: Remove this comment as well

File: .env [hunk 10]
```diff
@@ -19,14 +20,14 @@ PCRF_IP=172.22.0.4
...
 
 # SMF
 SMF_IP=172.22.0.7
 
 # UPF
 UPF_IP=172.22.0.8
-UPF_ADVERTISE_IP=172.22.0.8
+UPF_ADVERTISE_IP=172.22.0.8    # $DOCKER_HOST_IP
```
Comment (herlesupreeth) [inline .env:30]: Remove this comment as well

**Review by herlesupreeth — CHANGES_REQUESTED** [review]
Hey thanks again. I left some review comments to be addressed. Once done let me know I will merge it.

File: upf/upf.yaml (hunk 4 above)
Comment (b3c3ll) [reply upf/upf.yaml:13]: As we Exported .env file, there is no need to add "$"  in the .yaml files!
That was ok using docker compose V2.
Anyway, i'll do it!

**Review by herlesupreeth — APPROVED** [review]

**Comment by b3c3ll** [comment]
would you cherry-pick the commits from master branch to others ( specially hss_cx ) ?
Thanks
Best regards

**Comment by herlesupreeth** [comment]
Yep, just did that

**Pull request merged**